File: solutions/2520. Count the Digits That Divide a Number/2520.cpp
class Solution {
public:
    int countDigits(int num) {
        int count = 0;
        int temp = num;

        while (temp) {
            int val = temp % 10;
            if (num % val == 0) count++;
            temp /= 10;
        }
        return count;
    }
};

File: solutions/2520. Count the Digits That Divide a Number/2520.py
# Time complexity: O(d)
# Space complexity: O(1)
class Solution:
  def countDigits(self, num: int) -> int:
    ans = 0

    n = num
    while n > 0:
      if num % (n % 10) == 0:
        ans += 1
      n //= 10

    return ans